sdk/core/Azure.Core/samples/DynamicContent.md

```diff
@@ -0,0 +1,152 @@
+
+If no parameter is passed to `ToDynamicFromJson()`, properties names must exactly match the member names in the JSON content.  Passing `DynamicDataOptions.Default` will enable properties to be accessed using PascalCase property names, and will write any added properties with camelCase names.
+
+### Set a JSON property
+
+JSON members can be set on the dynamic object.
+
+```C# Snippet:AzureCoreSetDynamicJsonProperty
+Response response = await client.GetWidgetAsync("123");
+dynamic widget = response.Content.ToDynamicFromJson(DynamicDataOptions.Default);
+widget.Name = "New Name";
+```
+
+### Access an array value
+
+JSON array values are accessed using array indexers.
+
+```C# Snippet:AzureCoreGetDynamicJsonArrayValue
+Response response = await client.GetWidgetAsync("123");
+dynamic widget = response.Content.ToDynamicFromJson(DynamicDataOptions.Default);
+
+// JSON is "{ values = [1, 2, 3] }"
+int value = widget.Values[0];
+```
+
+### Check whether an optional property is present
+
+Optional properties will return null if not present in the JSON content.
+
+```C# Snippet:AzureCoreGetDynamicJsonOptionalProperty
+Response response = await client.GetWidgetAsync("123");
+dynamic widget = response.Content.ToDynamicFromJson(DynamicDataOptions.Default);
+
+// Check whether optional property is present
+if (widget.Properties != null)
+{
+    string color = widget.Properties.Color;
+}
+```
+
+### Enumerate a collection
+
+Dynamic JSON objects and arrays are `IEnumerable` and can be iterated over with the `foreach` keyword.
+
+```C# Snippet:AzureCoreEnumerateDynamicJsonObject
+Response response = await client.GetWidgetAsync("123");
+dynamic widget = response.Content.ToDynamicFromJson(DynamicDataOptions.Default);
+
+foreach (dynamic property in widget.Properties)
+{
+    UpdateWidget(property.Name, property.Value);
+}
+
+void UpdateWidget(string name, string value)
+{
+    Console.WriteLine($"Widget has property {name}='{value}'.");
+}
+```
+
+### Cast to a POCO type
+
+Dynamic JSON objects can be cast to CLR types using the cast operator.
+
+```C# Snippet:AzureCoreCastDynamicJsonToPOCO
+Response response = await client.GetWidgetAsync("123");
+dynamic content = response.Content.ToDynamicFromJson(DynamicDataOptions.Default);
+Widget widget = (Widget)content;
+```
+
+```C# Snippet:AzureCoreDynamicJsonPOCO
+public class Widget
+{
+    public string Name { get; set; }
+}
+```
+
+### Get a property with invalid C# characters in the name
+
+JSON members whose names have characters that are not valid for property names in C# can be accessed using property indexers.
+
+```C# Snippet:AzureCoreGetDynamicPropertyInvalidCharacters
+Response response = await client.GetWidgetAsync("123");
```

Comment:
It would be good if the JSON payload here actually had the required properties.

Reply:
I believe I have addressed this, but please tell me if you had something different in mind.